### ISSUE TITLE
pip vs pip3 in CONTRIBUTING.md

### ISSUE THREAD

**Issue by consideRatio** [jupyterhub/jupyterhub#2191]
There is a mix of using pip and pip3 in the CONTRIBUTING.md docs. I followed the instructions on a mac, and found the pip3 to fail but pip to work after installing miniconda according to instructions.

Go with `pip` and `python` over `pip3` and `python3`?

I'm confused about when `python3` is available btw, I'm also confused about other things =D See below. I'm ready to accept this confusion though.

![image](https://user-images.githubusercontent.com/3837114/46059081-03d24a80-c15e-11e8-8468-bad684efe953.png)

**Comment by betatim** [comment]
I think a good strategy is to consistently use `python3` and `pip3`. This means that inexperienced users notice they should use Python 3 (not Python 2). I think the difference is most relevant to people who use the Python bundled with their system. Everyone who is experienced enough to have created a setup with not-their-system-python should know that if they created a virtualenv/conda env/something else with just Python 3 in it that if `pip3` fails `pip` is the right answer. Maybe we can add a call out to mention that assumption?

**Comment by minrk** [comment]
Probably the most stable command is `python3 -m pip`? I don't know if/when pip3 doesn't work, that's strange to me. @consideRatio do you still have an env where `pip3` doesn't work? If so, can you `ls env/bin` and show `which pip` and `which pip3` and `conda list`?

> I'm confused about when python3 is available btw.

It should be always. I think assuming `python3` is available is a safe assumption for us, and if it's not true, there's something wrong that shouldn't be our responsibility, but maybe I'm mistaken.

**Comment by willingc** [comment]
I agree with @minrk that `python3 -m pip` is the most stable approach.

**Comment by yuvipanda** [comment]
I agree with @willingc  that `python3 -m pip` is the safest option. I'll amend docs accordingly!

Thanks for bringing this up, @consideRatio!
